### PR TITLE
[Reopen] Support scan key and value in meta

### PR DESCRIPTION
**Pull request by darionyaphet**
* Support scan key and value processor.
* Fix scan to support key and value .
* Add some small improve

### CODE REVIEW

File: src/interface/meta.thrift [hunk 1]
```diff
@@ -277,9 +277,15 @@ struct ScanReq {
 }
 
 struct ScanResp {
-    1: ErrorCode code,
-    2: common.HostAddr  leader,
-    3: list<string> values,
+    1: ErrorCode                                                 code,
+    2: common.HostAddr                                           leader,
+    3: map<string, string>(cpp.template = "std::unordered_map")  values,
```
Comment (dangleptr) [inline src/interface/meta.thrift]: rename it "Result" ?

File: src/interface/meta.thrift [hunk 2]
```diff
@@ -321,6 +327,8 @@ service MetaService {
     ExecResp remove(1: RemoveReq req);
     ExecResp removeRange(1: RemoveRangeReq req);
     ScanResp scan(1: ScanReq req);
+    KeyOrValueScanResp scanKey(1: ScanReq req);
```
Comment (dangleptr) [inline src/interface/meta.thrift]: Do we really need the two interfaces?

File: src/client/cpp/GraphClient.cpp [hunk 3]
```diff
@@ -4,7 +4,6 @@
  * attached with Common Clause Condition 1.0, found in the LICENSES directory.
  */
 
-#include "base/Base.h"
```
Comment (dangleptr) [inline src/client/cpp/GraphClient.cpp]: Why remove it?

File: src/common/base/Status.h [hunk 4]
```diff
@@ -98,10 +98,7 @@ class Status final {
     STATUS_GENERATOR(SyntaxError);
 
     // TODO(dangleptr) we could use ErrorOr to replace SpaceNotFound here.
-    STATUS_GENERATOR(SpaceNotFound);
```
Comment (dangleptr) [inline src/common/base/Status.h]: Why?

File: src/interface/meta.thrift (hunk 2 above)
Comment (darionyaphet) [reply src/interface/meta.thrift]: Actually, I considered use a flag to distinguish scan the keys and the values.

But I find it‘s confuse.

File: src/interface/meta.thrift (hunk 1 above)
Comment (darionyaphet) [reply src/interface/meta.thrift]: fixed

File: src/common/base/Status.h (hunk 4 above)
Comment (darionyaphet) [reply src/common/base/Status.h]: We have remove them from `meta.thrift`'s `ErrorCode` and replace with `E_NOT_FOUND`

File: src/client/cpp/GraphClient.cpp (hunk 3 above)
Comment (darionyaphet) [reply src/client/cpp/GraphClient.cpp]: duplicate imported from another `.h`

File: src/interface/meta.thrift (hunk 2 above)
Comment (dangleptr) [reply src/interface/meta.thrift]: I mean do we really need to scan key or value only? 
There is no efficient problems for the interfaces in meta.

File: src/common/base/Status.h (hunk 4 above)
Comment (dangleptr) [reply src/common/base/Status.h]: Why remove them?  Any benefits?

**Comment by whitewum** [comment]
is it still on going?

**Pull request closed without merging**